packages/core/src/navigation/navigation.element.ts

```diff
@@ -0,0 +1,479 @@
+  protected rootNavigationItems: NodeListOf<CdsNavigationItem>;
+
+  /**
+   * query for the root level start items and project them into the navigation-start slot.
+   */
+  @querySlot(':scope > cds-navigation-start', { assign: 'navigation-start' })
+  protected rootNavigationStart: CdsNavigationStart;
+
+  /**
+   * query for items inside a cds-navigation-group, used to pass state down
+   */
+  @querySlotAll(':scope > cds-navigation-group > cds-navigation-item')
+  protected navigationGroupItems: NodeListOf<CdsNavigationGroup>;
+
+  /**
+   * query for all cds-navigation elements, used to pass state down
+   */
+  @querySlotAll('cds-navigation-item')
+  protected navigationItemRefs: NodeListOf<CdsNavigationItem>;
+
+  /**
+   * query for all groups (including any nested groups), used ot pass state down
+   */
+  @querySlotAll('cds-navigation-group')
+  protected navigationGroupRefs: NodeListOf<CdsNavigationGroup>;
+
+  private initAriaActiveDescendant() {
+    /**
+     * If there is a currentActiveItem, focus on that
+     * If there is not a currentActiveItem focus on the first visible element
+     */
+    const focusElement = this.currentActiveItem ? this.currentActiveItem : this.allNavigationElements[0];
+    setFocus(focusElement);
+    this.ariaActiveDescendant = focusElement.id;
+  }
+
+  private blurActiveElement() {
+    if (this.currentActiveItem) {
+      removeFocus(this.currentActiveItem);
+    }
+  }
+
+  /**
+   * Rules for keyboard handling logic:
+   *
+   * 1. when cds-navigation element receives focus if there is already an active focus item,
+   *    set focus on it, else set focus on first focusable item
+   * 2. arrow key down sets focus on the next focusable item, if last item it moves focus to first focusable item
+   * 3. arrow key up sets focus on the previous focusable item, if first it moves focus to the last focusable item
+   * 4. arrow key left on a cds-navigation-item inside cds-navigation-group will put focus on the cds-navigation-start
+   *    button for the group
+   * 5. arrow key left on a cds-navigation-start element inside a cds-navigation-group will emit the groups
+   *    expandedChange event
+   * 6. arrow key right on a non expanded cds-navigation-group will emit the groups expandedChange event
+   * 7. arrow key left on a root cds-navigation-start element will fire the cds-navigation expandedChange event if
+   *    the cds-navigation element is expanded
+   * 8. arrow key right on a root cds-navigation-start element will fire the cds-navigation expandedChange event if
+   *    the cds-navigation element is not expanded
+   * 9. home key will move focus to the first focusable item
+   * 10. end key will move focus to the last focusable item
+   *
+   * We may need a way to let consumers mark elements and include them in the focusable elements, not sure how yet
+   *
+   * @param e
+   * @private
+   */
+  private keyboardNavigationHandler = (e: KeyboardEvent) => {
```

Comment:
This message has a lot of consecutive conditional statement inside the `onKey` handlers. Have you considered doing an early return or a switch statement if only one conditional is intended to pass?

Reply:
I'm adding returns to the onKey listeners with multiple branches here for now. I'd like to make it simpler but am not 100% sure how yet. Switch (imo) makes it more verbose and harder to follow. 

My thinking is at the event level